### ISSUE TITLE
Error on pre-rendered-html example `npm start` script

### ISSUE THREAD

**Issue by markpalfreeman** [HenrikJoreteg/hjs-webpack#205]
Today I tried swapping in `hjs-webpack` for a custom webpack config to take advantage of the pre-rendering helpers, so I started through the examples (not my project code) and ended with a console error.

Here are my steps:
```
git clone hjs-webpack
cd /pre-rendered-html-files
npm install
npm start
```
And the ensuing error message:
`Error: locals[0] does not appear to be a 'module' object with Hot Module replacement API enabled. You should disable react-transform-hmr in production by using 'env' section in Babel configuration.`

(Node: v5.10.0, npm: v3.8.3)

I've been reading a lot about @gaearon's [`react-transform-hmr`](https://github.com/gaearon/react-transform-hmr) package and its many variables, particularly in connection with Babel 6 and dev/production environments.

Has anyone else come across this or know how to fix it?

**Comment by HenrikJoreteg** [comment]
oooh, yeah this is not so fun. I think Dan's basically backed away from this approach. Personally i've been leaning more toward just good 'ol fashioned "dumb" hot loading. As long as all my state (including urls) are in redux outside of the "rendering tree" of react, you lose component state, but you can hotswap the whole thing. Dan is basically discontinuing it, it seems: https://medium.com/@dan_abramov/hot-reloading-in-react-1140438583bf#.i7b6yi8lf

All that said, if you want to see an example of a project that's set up this way and works, see this: https://github.com/henrikjoreteg/fluent-2016

**Comment by gaearon** [comment]
>Dan is basically discontinuing it

This particular project, yea, but I’m not giving up on hot reloading with state. It’ll just require a slighly different approach and some code to be written, as I wrote up in the article :wink:

**Comment by lukekarrys** [comment]
I've also hit this when using SSR and found that it doesn't play nice with `babel-core/register`, and I worked around it by removing the hmre config from the `.babelrc` and then adding the config back as the `query` part of the babel loader. Not perfect, but at least a workaround do get hmre working if you want to use it in this way.

[Here's an example from a project where I did that](https://github.com/lukekarrys/universal-static-instagram/blob/d6d7cba8169fb5704fd235aba19b4115ee31c08b/webpack.config.js#L38-L47)

I'll see if I can get the example working in the same way.

**Comment by markpalfreeman** [comment]
^ I think I did read that on a thread somewhere too, @lukekarrys.

Slightly aside from this case, I got the same error in the browser after building and deploying static files. Babel didn't seem to be mimicking the `NODE_ENV` (I know this is set in hjs-webpack)—I'm not sure it's the "best" solution, but preceding my `webpack` build script command with `BABEL_ENV=production` fixed that.

I just wanted to raise the issue with this repo in particular if others are looking for examples... I'm pretty new to debugging webpack issues, so thanks for the input!
